Named Commands

Henyahyah opened the pull request:
I added the rest of the commands from the other subsystems.

Resolves #42

Henyahyah committed:
adding commands

Henry deleted a blank line and then readded it to the code. he also did other stuff like naming commands and registering commands for the coral scoring device, cage climber, and elevator: o my nomie
     -james

CoderLiam926 reviewed (dismissed):
Looks a' okay to me Henyahyah!

CoderLiam926 committed:
Merge branch 'main' into named-commands

kjrokos commented:
Hi Henry, 

I don't think we needed all of the possible commands. What we needed was for the elevator controls to not be placeholder (wait) commands.

However, we don't want to just copy-paste the elevator commands, because then if we need to modify it we'd have to make the change twice, which has the issue that it's easy for them to get out of sync when you modify one but forget the other.

You can see with the `align to reef` and `align to source` commands, what we did was to define the command as a named command, and then reference it by name in the trigger. Because the name is just a string, we want to make sure it's a constant so that we don't accidentally mistype one of the strings.

I probably should have been more explicit about that in the text of the issue - hopefully this makes more sense!

CoderLiam926 committed:
Merge branch 'main' into named-commands

CoderLiam926 committed:
Naming Commands, Organizing Commands, and Fixing Commands.

Since Henry did the wrong thing, (while somehow managing to add stuff we needed), I have actually fixed the original problem now.

CoderLiam926 committed:
Merge branch 'main' into named-commands

kjrokos commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -79,30 +79,46 @@ public RobotContainer() throws IOException, Exception {
+        NamedCommands.registerCommand(Constants.SCORE_L2, elevator.elevatorL2().alongWith(Commands.waitUntil(() -> elevator.isElevatorAtL2()).andThen(NamedCommands.getCommand(Constants.CORAL_SCORING_POSITION))));
+        NamedCommands.registerCommand(Constants.SCORE_L3, elevator.elevatorL3().alongWith(Commands.waitUntil(() -> elevator.isElevatorAtL2()).andThen(NamedCommands.getCommand(Constants.CORAL_SCORING_POSITION))));
+        NamedCommands.registerCommand(Constants.SCORE_L4, elevator.elevatorL4().alongWith(Commands.waitUntil(() -> elevator.isElevatorAtL2()).andThen(NamedCommands.getCommand(Constants.CORAL_SCORING_POSITION))));
+        NamedCommands.registerCommand(Constants.EXTEND_ALGAE_REMOVER, elevator.dealgaeExtend());
+        NamedCommands.registerCommand(Constants.RETRACT_ALGAE_REMOVER, elevator.dealgaeRetract());
```
IIRC, we're no longer planning to have a motor extend or retract the algae arm - it will just be a mechanical linkage that extends when the elevator moves up. I think these can be removed.

CoderLiam926 replied:
Okie dokie!

CoderLiam926 replied:
I did it, but should I remove the rest of the stuff for the Dealgaer as well?

kjrokos replied:
no, there will still be powered wheels, so what you left is good

CoderLiam926 replied:
Oh, okay!

kjrokos commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -79,30 +79,46 @@ public RobotContainer() throws IOException, Exception {
+        controlTriggers.elevatorL4.onTrue(NamedCommands.getCommand(Constants.SCORE_L4));
+
+        controlTriggers.deAlgae.whileTrue(NamedCommands.getCommand(Constants.DEALGAE_ON));
+        reefTrigger.toggleOnTrue(NamedCommands.getCommand(Constants.OUTTAKE_CORAL));
```
I think this should just be `onTrue`

CoderLiam926 replied:
Change "toggleOnTrue"

CoderLiam926 replied:
Add 3ish second timer thingy

CoderLiam926 replied:
I changed it!

kjrokos commented on .vscode/settings.json:
```diff
@@ -56,5 +56,6 @@
     "edu.wpi.first.math.**.struct.*",
-  ]
+  ],
+  "java.jdt.ls.vmargs": "-XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx2G -Xms100m -Xlog:disable"
```
This is not relevant for this PR - we should remove it.

CoderLiam926 replied:
Oops

CoderLiam926 replied:
I reverted that now!

CoderLiam926 committed:
Removing Unneeded and Changing Input

I change an input from "toggleOnTrue" to "onTrue", I removed unneeded stuff for the Dealgaer, and I reverted a setting I accidentally changed somehow.

CoderLiam926 committed:
Merge branch 'main' into named-commands
